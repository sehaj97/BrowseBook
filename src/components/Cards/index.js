import React, { useEffect, useState } from 'react';

function Cards ({books, numFound}){
    return(
        <>
        {
        (numFound === 0) ? 
        (
            <h2 className="header">No results Found. Look up something else</h2>
        )
    :
        (
            <h2 className="header">{numFound} results Found</h2>
            
        )}
            { books && (<div className="row" > 
                {books.map((book, index) => (
                <div className="column" key={book.title + index}> 
                        <div className="card" > 
                         <h2> {book.title}</h2>
                         <img
                            src={book.isbn?`https://covers.openlibrary.org/b/isbn/${book.isbn[0]}-L.jpg`: require(`../../assets/images/nobook.jpg`)}
                            alt={book.title}
                            className="img-fit"
                            key={book.title + index}
                        />
                            {book.author_name && (<h3>By: {book.author_name[0]}</h3>)}
                            {book.first_publish_year && (<p>Published On: {book.first_publish_year}</p>)}
                            {book.isbn && (<p>isbn: {book.isbn[0]}</p>)}
                        </div>
                </div>
            ))}
            </div>)}
        </>
    )
}

export default Cards