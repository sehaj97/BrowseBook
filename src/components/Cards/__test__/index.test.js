import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Cards from '..';

afterEach(cleanup);

describe('Cards component', () => {
  it('renders', () => {
    render(<Cards />);
  });

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Cards />);
    expect(asFragment()).toMatchSnapshot();
  });

  })