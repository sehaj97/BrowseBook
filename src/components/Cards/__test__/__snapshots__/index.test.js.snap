// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <h2
    class="header"
  >
     results Found
  </h2>
</DocumentFragment>
`;
